pkg/login/social/azuread_oauth.go

```diff
@@ -130,6 +134,11 @@ func (s *SocialAzureAD) UserInfo(ctx context.Context, client *http.Client, token
 	}
 	s.log.Debug("AzureAD OAuth: extracted groups", "email", email, "groups", fmt.Sprintf("%v", groups))
 	if !s.isGroupMember(groups) {
+		if len(groups) == 0 {
+			// either they do not have a group or misconfiguration
+			return nil, errAzureADMissingGroups
+		}
+		// a group found in the claims is not in the allowed groups
```

Comment:
I think we can move this condition outside its outer if condition `!s.isGroupMember(groups)` because if `len(groups) == 0` there is no point in checking if the user is part of an empty list of groups, right?

Reply:
No it needs to be inside, if they have not configured required groups they should be able to sign in anyway